integration_test/third_party_apps_test.go

```diff
@@ -187,6 +190,34 @@ type expectedEntry struct {
 	FieldMatchers map[string]string `yaml:"field_matchers"`
 }
 
+// expectedMetricsEntry encodes a series of assertions about what data we expect
+// to see in the metrics backend.
+type expectedMetricEntry struct {
+	// The metric type, for example workload.googleapis.com/apache.current_connections.
+	Type string `yaml:"type"`
+	// The value type, for example INT64.
+	ValueType string `yaml:"value_type"`
+	// The kind, for example GAUGE.
+	Kind string `yaml:"kind"`
+	// The monitored resource, for example gce_instance.
+	// Currently we only test with gce_instance, so we expect
+	// all expectedMetricsEntries to have gce_instance.
+	MonitoredResource string `yaml:"monitored_resource"`
+	// Mapping of expected label keys to value patterns.
+	// Patterns are RE2 regular expressions.
+	Labels map[string]string `yaml:"labels"`
+	// If set to true, skip this entry during the test.
+	// Used for documenting metrics that exist in the backend
+	// but have not yet been implemented in this test, usually
+	// because ingesting them requires some additional complex
+	// setup.
```

Comment:
I think this part of the comment is somewhat misleading — sometimes a metric just isn't sent by the integration on all platforms…

Reply:
How about marking it sound more general:  some metrics are not present all the times. The reasons are:
The current test setup doesn’t emit such metrics
Some platform doesn’t emit such metrics
.
.
.

Reply:
I must've underestimated the ways that metrics can be missing. I'll fix this and related threads to treat missing metrics as "optional metrics are there sometimes" as opposed to "optional metrics are never there". (If we ever do need a "metric is never sent" field, we can always add a `skip: true` or something later on.)

Reply:
Updated wording.